Data/Functor/Foldable.hs

```diff
@@ -159,10 +156,10 @@ class Functor (Base t) => Recursive t where
   cata f = c where c = f . fmap c . project
 
   para :: (Base t (t, a) -> a) -> t -> a
-  para t = p where p x = t . fmap ((,) <*> p) $ project x
+  para f = p where p x = f . fmap ((,) <*> p) $ project x
 
-  gpara :: (Corecursive t, Comonad w) => (forall b. Base t (w b) -> w (Base t b)) -> (Base t (EnvT t w a) -> a) -> t -> a
-  gpara t = gzygo embed t
+  gpara :: Corecursive t => Gather (Base t) a s -> (Base t (t, s) -> a) -> t -> a
+  gpara = gzygo embed
```

Comment:
Should I keep the `t` like in the original version? It's not clear to me why it was eta-expanded in the first place.